lib/rubocop/cop/style/case_equality.rb

```diff
@@ -47,6 +54,18 @@ def const?(node)
             true
           end
         end
+
+        def replacement(lhs, rhs)
+          case lhs.type
+          when :regexp
+            "#{rhs.source} =~ #{lhs.source}"
```

Comment:
`/regexp/ === object` and `object =~ /regexp/` are not interchangeable.  The former returns true or false whereas the latter returns an index or nil.

Reply:
Exactly! I've opened the PR #8527.